queue_job/models/queue_job.py

```diff
@@ -281,10 +281,18 @@ def autovacuum(self):
                 ['|',
                  ('date_done', '<=', deadline),
                  ('date_cancelled', '<=', deadline),
-                 ('channel', '=', channel.complete_name)],
+                 ('channel', '=', channel.complete_name)], limit=1000
             )
-            if jobs:
+            while jobs:
                 jobs.unlink()
+                self._cr.commit()
+                jobs = self.search(
+                    ['|',
+                     ('date_done', '<=', deadline),
+                     ('date_cancelled', '<=', deadline),
+                     ('channel', '=', channel.complete_name)], limit=1000
+                )
```

Comment:
Store the domain before calling the first search, so you can reuse it here too

```suggestion
                jobs = self.search(domain, limit=1000)
```

Reply:
Or use this pattern:

```
while True:
    jobs = self.search(
        ['|',
         ('date_done', '<=', deadline),
         ('date_cancelled', '<=', deadline),
         ('channel', '=', channel.complete_name)], limit=1000
    )
    if not jobs:
        break
    jobs.unlink()
    self.env.cr.commit()
```